### PR TITLE
Deprecate aws_partition and use endpoint,regions instead

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
## What does this PR do?

This PR is to deprecate `aws_partition` config parameter and use `endpoint` with `regions` instead in order to make the AWS configuration less confusing. 

## Why is it important?

There is overlap between `aws_partition` and `endpoint`. We decide to use `endpoint` moving forward.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

1. Test case with no `endpoint` specified
Enable `aws` module:
```
./metricbeat modules enable aws
```

Change `aws.yml` config file to run `ec2` metricset:
```
- module: aws
  period: 5m
  credential_profile_name: elastic-beats
  metricsets:
    - ec2
```
This should work like it used to.

2. Test case with `endpoint`
```
- module: aws
  period: 5m
  credential_profile_name: elastic-beats
  endpoint: amazonaws.com
  regions: us-east-1
  metricsets:
    - ec2
```
Start Metricbeat and `ec2` metricset should work and collect metrics for instances in `us-east-1`.

## Related issues

Closes https://github.com/elastic/beats/issues/23153

### CODE REVIEW

File: x-pack/metricbeat/module/aws/aws.go [hunk 1]
```diff
@@ -92,6 +92,7 @@ func NewMetricSet(base mb.BaseMetricSet) (*MetricSet, error) {
 		Latency:       config.Latency,
 		AwsConfig:     &awsConfig,
 		TagsFilter:    config.TagsFilter,
+		Endpoint:      config.AWSConfig.Endpoint,
```
Comment (jsoriano) [inline x-pack/metricbeat/module/aws/aws.go:95]: Should we fail if both `Endpoint` and `AWSPartition` are provided?

File: x-pack/metricbeat/module/aws/_meta/docs.asciidoc [hunk 2]
```diff
@@ -23,7 +23,7 @@ Please see <<aws-credentials-config,AWS credentials options>> for more details.
 This module also accepts optional configuration `regions` to specify which
 AWS regions to query metrics from. If the `regions` parameter is not set in the
 config file, then by default, the `aws` module will query metrics from all available
-AWS regions.
+AWS regions. If `endpoint` is specified, `regions` becomes a required config parameter.
```
Comment (jsoriano) [inline x-pack/metricbeat/module/aws/_meta/docs.asciidoc:26]: I'd suggest to add a section about connecting to AWS partitions, with an example of the settings needed to connect with one of them using `endpoint` and `region`.

**Review by jsoriano — APPROVED** [review]
Looks good, added a couple of small suggestions.

File: x-pack/metricbeat/module/aws/_meta/docs.asciidoc (hunk 2 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/aws/_meta/docs.asciidoc:26]: Good point! I will add a section for AWS partitions! Thank you for the review!

**Comment by andresrc** [comment]
Do we have an issue to remove the deprecated parameter in 8.0?

**Comment by kaiyan-sheng** [comment]
@andresrc I just created an issue for removing the config in 8.0.0: https://github.com/elastic/beats/issues/23590

File: x-pack/metricbeat/module/aws/aws.go (hunk 1 above)
Comment (jsoriano) [reply x-pack/metricbeat/module/aws/aws.go:96]: Yep, you are right, I was thinking on `endpoint` as a new setting, but it existed before.

Would it make sense to log a warning when both are used mentioning the endpoint and region that are going to be used?

**Review by jsoriano — APPROVED** [review]

File: x-pack/metricbeat/module/aws/aws.go (hunk 1 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/aws/aws.go:96]: Done :) Thanks!

**Pull request merged**